sys-apps/ipmicfg: New package

ConiKost opened the pull request:
sys-apps/ipmicfg: New package

Closes: https://bugs.gentoo.org/601318

mgorny commented on sys-apps/ipmicfg/ipmicfg-1.27.1.170901.ebuild:
```diff
@@ -0,0 +1,59 @@
+
+EAPI=6
+
+inherit versionator
```
Please use `eapi7-ver` instead (it's forward-compat for EAPI 7 version functions).

ConiKost replied:
Changed to eapi7-ver by newest push.

mgorny commented on sys-apps/ipmicfg/ipmicfg-1.27.1.170901.ebuild:
```diff
@@ -0,0 +1,59 @@
+
+pkg_nofetch() {
+	elog "Please download ${A} from"
+	elog "${HOMEPAGE}/SwDownload/UserInfo.aspx?sw=0&cat=IPMI"
```
Please don't ref HOMEPAGE like this. It is multi-value, and people may want to add extra sites to it.

ConiKost replied:
What to do instead? Write the full plain URL instead of a variable?

mgorny replied:
Yes.

ConiKost replied:
URL is now in plain text by newest push.

mgorny commented on sys-apps/ipmicfg/ipmicfg-1.27.1.170901.ebuild:
```diff
@@ -0,0 +1,59 @@
+src_install() {
+	# Choose ARCH
+	if use amd64; then
+		local MY_ARCH_BINARY="x86_64"
```
Local variables go lowercase in bash.

ConiKost replied:
Fixed by newest push.

mgorny commented on sys-apps/ipmicfg/ipmicfg-1.27.1.170901.ebuild:
```diff
@@ -0,0 +1,59 @@
+	dosym "../ipmicfg/IPMICFG-Linux.${MY_ARCH_BINARY}" "/opt/bin/ipmicfg"
+
+	# Install docs
+	local DOCS="IPMICFG_UserGuide.pdf ReleaseNotes.txt"
```
Array, please.

ConiKost replied:
Whoops, sorry. Fixed by newest push.

Zlogene commented on sys-apps/ipmicfg/ipmicfg-1.27.1.170901.ebuild:
```diff
@@ -0,0 +1,59 @@
+MY_PN="${PN^^}"
+MY_PV="$(ver_cut 1-3)"
+
+DESCRIPTION="IPMICFG is an in-band utility for configuring IPMI devices"
```
`DESCRIPTION` should not contain a package name

ConiKost replied:
Changed by newest push.

Zlogene commented on sys-apps/ipmicfg/ipmicfg-1.27.1.170901.ebuild:
```diff
@@ -0,0 +1,59 @@
+
+DESCRIPTION="IPMICFG is an in-band utility for configuring IPMI devices"
+HOMEPAGE="https://www.supermicro.com"
+SRC_URI="ftp://ftp.supermicro.com/utility/${MY_PN}/${MY_PN}_${MY_PV}_build.${MY_DATE}.zip"
```
add `app-arch/unzip` to `DEPEND` as `SRC_URI` pointed to the `.zip` archive

ConiKost replied:
Added by newest push.

ConiKost commented:
@mgorny: I've changed the QA_PREBUILT. Do you know, how I do reproduce this QA warning? You told me yesterday about "-frecord-gcc-switches", which I have added to my CFLAGS, but still, I didn't got this warning.

mgorny commented:
Looks good now, thanks!
